### DISCUSSION TITLE
Str::studly() give back Too Few Arguments after update.

### DISCUSSION THREAD

**Discussion by theoreaves** [livewire/livewire#5600] [1. Help] [answered]
My project just updated to livewire version 2.12.2, and now it is randomly giving out this error:

```Too few arguments to function Illuminate\Support\Str::studly(), 0 passed in /home/tbsweb/sites/dyersburg/vendor/livewire/livewire/src/helpers.php on line 13 and exactly 1 expected {"userId":239,"exception":"[object] (ArgumentCountError(code: 0): Too few arguments to function Illuminate\\Support\\Str::studly(), 0 passed in /home/********/vendor/livewire/livewire/src/helpers.php on line 13 and exactly 1 expected at /home/********/vendor/laravel/framework/src/Illuminate/Support/Str.php:1109)```

Only vendor pages are returned int he error stack, no application pages. I not sure where exactly to look.  I've tried this:

`composer dump-autoload`
`php artisan cache:clear`
`php artisan config:clear`
`php artisan view:clear`

It seemed to help, but some users still see the error on occasion.  I think maybe the users that have cleared their browser cache also are no longer getting the error, but I can't confirm that.

Any suggestions where to look?

**Comment by joshhanley** [comment]
@theoreaves can you share the full stacktrace? It would be good to know where `Str::studly()` is being called from. Also is it just one component or all of them causing issues?

**Reply by theoreaves** [reply]
Here is a link to the full stack:
[https://flareapp.io/share/Bm0xlx6m#F58](Error)

And yes, I've only seen this on one component.  I did go back to a previous version (2.10.7) and the error doesn't occur there.

Thanks!

**Comment by joshhanley** [comment, marked as answer]
@theoreaves thanks! Not sure what changed to make it noticeable. But it seems you have an empty wire:model, see how the name is null? Look for `wire:model=''` or just `wire:model` on your input and add the property name to it. 

![B3D15DF8-EBC0-4AD8-BE51-C3D0A26E8D3E](https://user-images.githubusercontent.com/882837/222821481-a678ba4b-5e40-4e38-8cf8-413cc6f99e9b.jpeg)

Hope this helps!

**Comment by theoreaves** [comment]
So, 

What I figured out is that field's wire:model was wire:model.defer.  and the screen would reload at a point with a different component that did not have defer, but lazy.  I'm guessing when the field reload and that changes, it must cause a problem?

I was able to just change both versions of the field to wire:model.lazy and that fixed the issue I was having.

Thanks!

**Reply by joshhanley** [reply]
@theoreaves ah ok! No worries 🙂 Sounds like a DOM diffing issue. Make sure to use wire:key if you use conditionals or loops in your blade template, which stops Livewire getting confused as to which element is what.

**Reply by svenhorner** [reply]
I came here, cause I had the same error message, in my case it was due to duplicate id on two different input components. What puzzled me for a moment was, that it would work with the duplicate id as long as I had the same debounce time set for both.
